### PR TITLE
Cleanup & preparation for further changes

### PR DESCRIPTION
**Pull request by eemeli**
These are preparatory steps for making the code easier to work with going forward, applying a consistent code style, adding type hints (incomplete, but a start), and updating deprecated code.

~The first commit is a result of running `black` on all `.py` files, so reviewing the other changes is probably easier done commit by commit.~

### CODE REVIEW

**Comment by eemeli** [comment]
Rebased to split out the black styling into the now merged #7.

File: fluent/migrate/_context.py [hunk 1]
```diff
@@ -23,7 +26,11 @@ class InternalContext:
     For the public interface, see `context.MigrationContext`.
     """
 
-    def __init__(self, lang, reference_dir, localization_dir, enforce_translated=False):
+    dependencies: Dict[Tuple[str, str], Set[Tuple[str, Source]]] = {}
+    localization_dir: str
+    reference_dir: str
+
+    def __init__(self, lang, enforce_translated=False):
```
Comment (bcolsson) [inline fluent/migrate/_context.py:33]: You're adding annotations, but `localization_dir` and `reference_dir` are no longer being initialized anywhere within this class, even though they are being used by some of the methods.

From what I can tell, they might not be required to be initiated since they also get initiated in `context.MigrationContext`, but this was added in the commit for adding annotations, and this seems like a different sort of change.

Comment (eemeli) [reply fluent/migrate/_context.py:33]: They weren't initialized here before either; note how the parameters aren't actually used in the `__init__()` body. This class is not meant to ever be used except via `MigrationContext`, so its main function is to slightly obscure some of the context methods from the public API.

The two classes should probably be merged to simplify the situation somewhat, but that's a separate larger change that I've left out at least for now.

File: fluent/migrate/validator.py [hunk 2]
```diff
@@ -19,12 +19,14 @@ class BadContextAPIException(Exception):
 
 
 def process_assign(node, context):
-    if isinstance(node.value, ast.Str):
-        val = node.value.s
+    if isinstance(node.value, ast.Constant):
```
Comment (flodolo) [inline fluent/migrate/validator.py:22]: I'm lost here, and I feel like I'm missing something obvious. If I try locally, I see that `ast` has `ast.Constant`, but where the heck is that coming from?

```
>>> import ast
>>> print(ast.Constant)
<class 'ast.Constant'>
>>> vars(ast.Constant)
mappingproxy({'_fields': ('value', 'kind'), '__match_args__': ('value', 'kind'), '__module__': 'ast', '__doc__': 'Constant(constant value, string? kind)', 'kind': None, 'n': <property object at 0x104fc9cb0>, 's': <property object at 0x104fca070>})
```

Also, why this change, given Fluent hasn't changed in a while?

Comment (eemeli) [reply fluent/migrate/validator.py:22]: This change is due to `ast.Str` being [deprecated](https://bugs.python.org/issue26146) in all the versions of Python we now support, and is due to be removed in a near-future release:
> Changed in version 3.8: Class [ast.Constant](https://docs.python.org/3/library/ast.html#ast.Constant) is now used for all constants.

The `isinstance()` usage works because `ast.Str` [inherits](https://github.com/python/cpython/blob/5d7b90db0cd463e69d15acf5498232b9bc2bfeaa/Lib/ast.py#L594) from `ast.Constant`.

> Also, why this change, given Fluent hasn't changed in a while?

Because this is about parsing Python, and it is constantly changing.

Comment (flodolo) [reply fluent/migrate/validator.py:22]: Thanks, I missed this was Python's ast and not Fluent's.

**Review by flodolo — APPROVED** [review]

File: fluent/migrate/_context.py (hunk 1 above)
Comment (bcolsson) [reply fluent/migrate/_context.py:33]: 🤦 You're right - thanks for responding.

**Review by bcolsson — APPROVED** [review]
